src/main/java/cuchaz/enigma/utils/Message.java

```diff
@@ -0,0 +1,390 @@
+package cuchaz.enigma.utils;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.util.Objects;
+
+import cuchaz.enigma.network.packet.PacketHelper;
+import cuchaz.enigma.translation.representation.entry.Entry;
+
+public abstract class Message {
+
+	public final String user;
+	
+	public static Chat chat(String user, String message) {
+		return new Chat(user, message);
+	}
+
+	public static Connect connect(String user) {
+		return new Connect(user);
+	}
+
+	public static Disconnect disconnect(String user) {
+		return new Disconnect(user);
+	}
+
+	public static EditDocs editDocs(String user, Entry<?> entry) {
+		return new EditDocs(user, entry);
+	}
+
+	public static MarkDeobf markDeobf(String user, Entry<?> entry) {
+		return new MarkDeobf(user, entry);
+	}
+
+	public static RemoveMapping removeMapping(String user, Entry<?> entry) {
+		return new RemoveMapping(user, entry);
+	}
+
+	public static Rename rename(String user, Entry<?> entry, String newName) {
+		return new Rename(user, entry, newName);
+	}
+
+	public abstract String translate();
+
+	public abstract Type getType();
+
+	public static Message read(DataInput input) throws IOException {
+		byte typeId = input.readByte();
+		if (typeId < 0 || typeId >= Type.values().length) throw new IOException(String.format("Invalid message type ID %d", typeId));
```

Comment:
This line is getting a bit long, maybe put the if body on the next line (in braces)?